internal/exec/util/blkid.c

```diff
@@ -15,24 +15,200 @@
 #include <blkid/blkid.h>
 #include "blkid.h"
 
-result_t blkid_lookup(const char *device, const char *field_name, char buf[], size_t buf_len)
+static result_t extract_part_info(blkid_partition part, struct partition_info *info);
+
+static result_t checked_copy(char *dest, const char *src, size_t len);
+
+// checked_copy is a helper for extract_part_info. It checks if src is null and fits in dst.
+// It's basically a safer strncpy to reduce boilerplate
+// Returns:
+//   RESULT_LOOKUP_FAILED if src is null
+//   RESULT_OVERFLOW if src is too big
+//   0 on success
+static result_t checked_copy(char *dest, const char *src, size_t len) {
```

Comment:
Mostly because dealing with any sort of dynamically allocated data with CGO is a giant pain. We only care about gpt which is limited to 36 UTF16 characters, so the statically allocated buffers are fine.